db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/StoreDataManagerProvider.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2011-2020 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.infinispan.data;
+
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.data.StandaloneStoreDataManager;
+import org.commonjava.indy.data.StoreDataManager;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class StoreDataManagerProvider {
+
+    @Inject
+    StoreDataManager storeDataManager;
+
+    @Inject
+    @StandaloneStoreDataManager
+    InfinispanStoreDataManager infinispanStoreDataManager;
+
+    @Inject
+    IndyConfiguration indyConfiguration;
+
+    private boolean isStandaloneMode;
+
+    @PostConstruct
+    public void init()
+    {
+        isStandaloneMode = indyConfiguration.isStandalone();
```

Comment:
This isStandalone() is a bit overloaded. It was invented mainly to use disk-storage on indy static instances. For other instances, the isStandalone are false, but they use the ispnstoremanager. I suggest to add another config like "store.manager" to control it.

Reply:
Yeah, agree. It is possible that we will not apply the pathmap and other cassandra related stuff on prod at the same time, so using the isStandalone for these all is hard, it's better to use the specific one for storeDataManager instead.